tensorflow/stream_executor/cuda/cuda_blas.cc

```diff
@@ -25,6 +25,10 @@ limitations under the License.
 #define EIGEN_HAS_CUDA_FP16
 #endif
 
+#if CUDA_VERSION >= 8000
+#define CUBLAS_DATA_HALF CUDA_R_16F
+#endif
+
```

Comment:
This seems fine with me too. 

The downside is that it introduces an outdated symbol back into CUBLAS space, which might interact with other code when we move it around. I would second vrv@'s suggestion that it is better to have our own symbol that is conditionally defined.

I will leave it up to you whether you want to follow through in this CL.